### DISCUSSION TITLE
Pester Test - PowerShell

### DISCUSSION THREAD

**Discussion by Manvendra11** [pester/Pester#2079] [Q&A] [answered]
Hi @dlwyatt 

I have a module ABC.psm1

In that module there are multiple functions 
function a {}
function b {}
function c {}

i want to create a test that will check each function in the module and see if it has help file, is param block declared , etc etc. 

how can i go to scan each function using peste ?

**Comment by dlwyatt** [comment]
Dunno.  Probably some combination of Get-Command, the AST, and possibly Get-Help.  Once you have objects that contain all the things you're looking for, putting them into Pester tests with `Should` assertions is straightforward enough.

**Comment by fflaten** [comment, marked as answer]
Take a look at tst/Help.Tests.ps1 and tst/Pester.Tests.ps1 in this repo. We have a few tests like that.

**Comment by Manvendra11** [comment]
OK. Thank You All.
